src/Formatter/FullHttpMessageFormatter.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Http\Message\Formatter;
+
+use Http\Message\Formatter;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A formatter that prints the complete HTTP message.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class FullHttpMessageFormatter implements Formatter
+{
+    /**
+     * The maximum length of the body.
+     *
+     * @var int
+     */
+    private $maxBodyLendth;
+
+    /**
+     * @param int $maxBodyLendth number of chars.
+     */
+    public function __construct($maxBodyLendth = 1000)
+    {
+        $this->maxBodyLendth = $maxBodyLendth;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function formatRequest(RequestInterface $request)
+    {
+        $message = sprintf(
+            "%s %s HTTP/%s\n",
+            $request->getMethod(),
+            $request->getRequestTarget(),
+            $request->getProtocolVersion()
+        );
+
+        foreach ($request->getHeaders() as $name => $values) {
+            $message .= $name.': '.implode(', ', $values)."\n";
+        }
+
+        $message .= "\n".mb_substr($request->getBody()->__toString(), 0, $this->maxBodyLendth);
```

Comment:
What is the difference? Sure one may think one **looks** better than the other. But is one faster then the other? String casting of an object calls __toString() as far as I know. 

I like calling __toString but Im happy to change if the other way is better or more coherent with the rest of our code.